docker/etl/Dockerfile

```diff
@@ -0,0 +1,28 @@
+# syntax=docker/dockerfile:1
+FROM python:3.10.7-slim
+
+# install nix for data extraction
+RUN apt-get update && apt-get install -y curl
+RUN curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install linux --init none --no-confirm
+
+# set environment variables
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+ENV POETRY_VIRTUALENVS_CREATE=0
+ENV PATH="${PATH}:/nix/var/nix/profiles/default/bin"
+
+# install poetry
+RUN pip install "poetry==1.5.1"
+
+# install dependencies
+ADD ./core /code/core/
+ADD ./api/pyproject.toml ./api/poetry.lock /code/api/
+WORKDIR /code/api/
+RUN poetry install --with "local" --without "dev" --no-root
```

Comment:
> we may also remove [...] the `local` group in the pyproject.toml setup

The reason I've introduced `local` is due to the definition of the dependency on core

```toml
[tool.poetry.dependencies]
python = "^3.10"
...
nixpkgs-graph-core = { version = "^1.0.0" }
...

[tool.poetry.group.local.dependencies]
nixpkgs-graph-core = { path="../core" }
```

Without the "local" group and `--without dev`, poetry will try to find `nixpkgs-graph-core` in PyPI.
And we don't want to specify a local dependency in the main dependencies because that lock deployment/distribution.

Reply:
I see. Thanks for the clarification @GuillaumeDesforges !